macOS app bundle error related to jcon during build

This is related to #667. @mobbyg raised this issue again via Discord, so I figured it'd be good to get back on our radar. @broxen and I researched this issue for a while about a month ago. Neither of us have a working system running macOS readily available to test fixes on however. It can be tedious to push potential solutions via git and wait for Travis to build each time. If @mobbyg cannot troubleshoot this, hopefully someone else with a working system can. Otherwise, we can use something like https://github.com/DrDonk/unlocker.

```
[100%] Built target jcon_ex
/usr/local/Cellar/cmake/3.10.0/bin/cmake -E cmake_progress_start /Users/travis/build/Segs/Segs/build/3rd_party/jcon_BUILD-prefix/src/jcon_BUILD-build/CMakeFiles 0
/Applications/Xcode-9.2.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 preinstall
make[4]: Nothing to be done for `preinstall'.
Install the project...
/usr/local/Cellar/cmake/3.10.0/bin/cmake -P cmake_install.cmake
-- Install configuration: "Debug"
-- fixup_bundle
--   app='/Users/travis/build/Segs/Segs/build/3rd_party/built/jcon_ex.app'
--   libs=''
--   dirs=''
--   ignoreItems=''
-- warning: *NOT* handled - directory/file does not exist...
CMake Error at /usr/local/Cellar/cmake/3.10.0/share/cmake/Modules/BundleUtilities.cmake:966 (message):
  error: fixup_bundle: not a valid bundle
Call Stack (most recent call first):
  src/cmake_install.cmake:38 (fixup_bundle)
  cmake_install.cmake:37 (include)
-- fixup_bundle: done
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/lib/libjcon.a
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/jcon.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/jcon_assert.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_client.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_debug_logger.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_endpoint.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_error.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_file_logger.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_logger.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_request.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_result.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_server.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_socket.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_success.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_tcp_client.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_tcp_server.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_tcp_socket.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_websocket.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_websocket_client.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/json_rpc_websocket_server.h
-- Installing: /Users/travis/build/Segs/Segs/build/3rd_party/built/include/jcon/string_util.h
make[3]: *** [install] Error 1
make[2]: *** [3rd_party/jcon_BUILD-prefix/src/jcon_BUILD-stamp/jcon_BUILD-install] Error 2
make[1]: *** [3rd_party/CMakeFiles/jcon_BUILD.dir/all] Error 2
make: *** [all] Error 2
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -64,11 +64,10 @@ matrix:
       compiler: "gcc-7"
       env: BUILD='Debug' CC=gcc-7 CXX=g++-7
       addons: *gcc7
-# Disabling macOS builds temporarily until the app bundle issue can be resolved
-#   - os: osx
-#     osx_image: xcode9.2
-#     compiler: clang
-#     env: BUILD='Debug'
+    - os: osx
+      osx_image: xcode9.2
+      compiler: clang
+      env: BUILD='Debug'
 
 before_install:
   - if [ "$TRAVIS_OS_NAME" = "linux" ]; then sudo add-apt-repository ppa:beineri/opt-qt593-trusty -y; fi

--- a/3rd_party/jcon-cpp/src/CMakeLists.txt
+++ b/3rd_party/jcon-cpp/src/CMakeLists.txt
@@ -26,21 +26,21 @@ if(USE_QT)
     find_program(QT_UIC_EXECUTABLE uic)
 endif()
 
-if(APPLE)
-    # icon
-    if(EXISTS "${PROJECT_NAME}.icns")
-        # set how it shows up in the Info.plist file
-        set(MACOSX_BUNDLE_ICON_FILE ${PROJECT_NAME}.icns)
-
-        # set where in the bundle to put the icns file
-        set_source_files_properties(${PROJECT_NAME}.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
-
-        # include the icns file in the target
-        set(${PROJECT_NAME}_sources ${${PROJECT_NAME}_sources} ${PROJECT_NAME}.icns)
-    endif()
-
-    add_executable(${PROJECT_NAME} MACOSX_BUNDLE ${${PROJECT_NAME}_headers} ${${PROJECT_NAME}_sources} ${UI_HEADERS})
-elseif(WIN32)
+#if(APPLE)
+#    # icon
+#    if(EXISTS "${PROJECT_NAME}.icns")
+#        # set how it shows up in the Info.plist file
+#        set(MACOSX_BUNDLE_ICON_FILE ${PROJECT_NAME}.icns)
+#
+#        # set where in the bundle to put the icns file
+#        set_source_files_properties(${PROJECT_NAME}.icns PROPERTIES MACOSX_PACKAGE_LOCATION Resources)
+#
+#        # include the icns file in the target
+#        set(${PROJECT_NAME}_sources ${${PROJECT_NAME}_sources} ${PROJECT_NAME}.icns)
+#    endif()
+#
+#    add_executable(${PROJECT_NAME} MACOSX_BUNDLE ${${PROJECT_NAME}_headers} ${${PROJECT_NAME}_sources} ${UI_HEADERS})
+if(WIN32)
     add_executable(${PROJECT_NAME} WIN32 ${${PROJECT_NAME}_headers} ${${PROJECT_NAME}_sources} ${UI_HEADERS})
 else()
     add_executable(${PROJECT_NAME} ${${PROJECT_NAME}_headers} ${${PROJECT_NAME}_sources} ${UI_HEADERS})
@@ -53,23 +53,23 @@ if(USE_QT)
     qt5_use_modules(${PROJECT_NAME} Core Network Test WebSockets Widgets)
 endif()
 
-if(APPLE)
-    set(APPS "\${CMAKE_INSTALL_PREFIX}/bin/${PROJECT_NAME}")
-
-    set(plugin_dest_dir ${PROJECT_NAME}.app/Contents/MacOS)
-    set(APPS "\${CMAKE_INSTALL_PREFIX}/${PROJECT_NAME}.app")
-
-    install(CODE "
-            include(BundleUtilities)
-            fixup_bundle(\"${APPS}\" \"\" \"${DIRS}\")
-            " COMPONENT Runtime)
-
-else()
+#if(APPLE)
+#    set(APPS "\${CMAKE_INSTALL_PREFIX}/bin/${PROJECT_NAME}")
+#
+#    set(plugin_dest_dir ${PROJECT_NAME}.app/Contents/MacOS)
+#    set(APPS "\${CMAKE_INSTALL_PREFIX}/${PROJECT_NAME}.app")
+#
+#    install(CODE "
+#            include(BundleUtilities)
+#            fixup_bundle(\"${APPS}\" \"\" \"${DIRS}\")
+#            " COMPONENT Runtime)
+#
+#else()
     install(TARGETS ${EXECUTABLE_NAME}
         RUNTIME DESTINATION bin
         LIBRARY DESTINATION lib
         ARCHIVE DESTINATION lib/static)
-endif()
+#endif()
 
 if(MINGW)
     get_filename_component(MINGW_DIR ${CMAKE_C_COMPILER} PATH)

--- a/Projects/CoX/Servers/GameDatabase/GameDBSyncContext.cpp
+++ b/Projects/CoX/Servers/GameDatabase/GameDBSyncContext.cpp
@@ -124,7 +124,7 @@ bool GameDbSyncContext::loadAndConfigure()
 
     if(!m_db->open())
     {
-        qFatal("Failed to open database: %s", dbname);
+        qFatal("Failed to open database: %s", dbname.toStdString().c_str());
         return false;
     }
 

